### PR TITLE
feat(job): Simplify gcluster job config workflow

### PR DESCRIPTION
**Pull request by agrawalkhushi18**
### Description
This PR Improves the user experience and flexibility of managing persistent CLI configuration defaults (`project`, `cluster`, and `location`) for `gcluster job` commands.

### Key Changes
1. **Interactive System Editor Mode (`gcluster job config set`)**:
   - Running `./gcluster job config set` without arguments now opens the configuration file (`~/.gcluster/context.json`) in the user's default text editor (`$EDITOR`, falling back to `nano`).
   - Automatically initializes an empty template if the file does not yet exist.
   - Validates JSON structure on save with `decoder.DisallowUnknownFields()` to prevent syntax errors or misspelled configuration keys.
2. **Single Property Updates (`gcluster job config set [key] [value]`)**:
   - Allows setting individual defaults directly without opening an editor (e.g., `./gcluster job config set project my-project`).
   - Validates that values are non-empty and keys match supported fields (`project`, `cluster`, `location`).
3. **`gcluster job config show` Command**:
   - Added `gcluster job config show` to display the active configuration file path alongside a formatted `KEY / VALUE` table of currently saved defaults.
4. **Documentation & Unit Tests**:
   - Updated [README.md](README.md) and [docs/gcluster_job_guide.md](docs/gcluster_job_guide.md) with updated usage instructions and examples.
   - Added comprehensive unit tests in `cmd/job/config_test.go` and `cmd/job/context_test.go` covering edge cases, invalid keys/values, missing arguments, and corrupt files.
---

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: cmd/job/config.go [hunk 1]
```diff
@@ -34,29 +34,50 @@ var ConfigCmd = &cobra.Command{
...
-		switch key {
-		case "project":
-			ctx.ProjectID = value
-		case "cluster":
-			ctx.ClusterName = value
-		case "location":
-			ctx.Location = value
-		default:
-			return fmt.Errorf("invalid configuration key: %s. Supported keys: project, cluster, location", key)
+		isOldFormatKey := false
```
Comment (SwarnaBharathiMantena) [inline cmd/job/config.go]: The `isOldFormatKey` variable is not required here before deciding how to parse the arguments because setContextField() already acts as the gatekeeper for valid keys. 

Is `isSuspiciousValue` required? We are supporting a user by adding this and we are adding complexity to the code.

File: cmd/job/config_test.go [hunk 2]
```diff
@@ -42,3 +37,63 @@ func TestConfigSetCmd_NoGlobalFlagsRequired(t *testing.T) {
...
+func TestConfigSetCmd_LegacyBug(t *testing.T) {
+	tempDir := t.TempDir()
+
+	t.Setenv("HOME", tempDir)
+
+	output, err := executeCommand(JobCmd, "config", "set", "project", "cluster=foo")
+	if err == nil {
+		t.Fatalf("expected config set to fail with invalid argument format for suspicious value, but it succeeded: %s", output)
+	}
+}
```
Comment (SwarnaBharathiMantena) [inline cmd/job/config_test.go:130]: As Go heavily favors table-driven testing, we can consolidate these new scenarios into a single TestConfigSetCmd function using a slice of structs defining the args, expectedCtx, and expectErr. This will make it easier to add new edge cases in the future.

File: cmd/job/config.go (hunk 1 above)
Comment (agrawalkhushi18) [reply cmd/job/config.go]: Agreed! We simplified the design and removed those heuristic checks (isOldFormatKey and isSuspiciousValue) along with the batch parsing.

Now gcluster job config set simply accepts standard [key] [value] arguments (e.g., gcluster job config set project my-project) with direct validation, or opens the system editor interactively when run without arguments.

File: cmd/job/config_test.go (hunk 2 above)
Comment (agrawalkhushi18) [reply cmd/job/config_test.go:130]: Done! Refactored TestConfigSetCmd into a table-driven test using []struct { name, args, expectedCtx, expectErr }. It now cleanly tests each scenario (empty values, single argument, invalid keys, and valid assignments for project, cluster, and location) in a single test loop.

**Review by SwarnaBharathiMantena — DISMISSED** [review]

**Review by SwarnaBharathiMantena — APPROVED** [review]
